From b79101ccdbb609e3dccfe097952ab2d3d7b55951 Mon Sep 17 00:00:00 2001
From: Emmanuel Gil Peyrot <linkmauve@linkmauve.fr>
Date: Fri, 10 Jul 2020 03:39:46 +0200
Subject: [PATCH 03/18] Remove deprecated stock messages

---
 src/gui/gtk-lshw.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/gui/gtk-lshw.c b/src/gui/gtk-lshw.c
index 50dfb8329a5c..9ac525c715d8 100644
--- a/src/gui/gtk-lshw.c
+++ b/src/gui/gtk-lshw.c
@@ -31,8 +31,8 @@ main (int argc, char *argv[])
 				GTK_BUTTONS_NONE,
 				"Executing this program as a normal user will give incomplete and maybe erroneous information.");
     gtk_dialog_add_buttons(GTK_DIALOG(dialog),
-                                  GTK_STOCK_QUIT, GTK_RESPONSE_CANCEL,
-                                  GTK_STOCK_EXECUTE, GTK_RESPONSE_ACCEPT,
+                                  "_Quit", GTK_RESPONSE_CANCEL,
+                                  "_Execute", GTK_RESPONSE_ACCEPT,
                                   NULL);
 
     proceed = (gtk_dialog_run (GTK_DIALOG (dialog)) == GTK_RESPONSE_ACCEPT);
-- 
2.17.1

